hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCpOptions.java

```diff
@@ -234,7 +236,18 @@ public Path getSourceFileListing() {
 
   public List<Path> getSourcePaths() {
     return sourcePaths == null ?
-        null : Collections.unmodifiableList(sourcePaths);
+        null :
+        Collections.unmodifiableList(getUniquePaths(sourcePaths));
+  }
+
+  private List<Path> getUniquePaths(List<Path> srcPaths) {
+    Set<Path> uniquePaths = new LinkedHashSet<>();
+    for (Path path : srcPaths) {
+      if (!uniquePaths.add(path)) {
+        LOG.warn("Path: {} added multiple times, ignoring the redundant entry.", path);
```

Comment:
Should it log at info? I'm just worried because WARN Messages are often interpreted as a sign of serious problems; if this is happening due to some workflow then it's not going to go away.

Reply:
Yes, makes sense.
Thanx @steveloughran for the review. I have changed it to INFO